engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -76,11 +78,11 @@ public void processRecord(final TypedRecord<ProcessInstanceMigrationRecord> comm
         processState.getProcessByKeyAndTenant(
             targetProcessDefinitionKey, processInstance.getValue().getTenantId());
     if (processDefinition == null) {
-      // todo: we should reject the command explicitly
-      throw new IllegalStateException(
-          String.format(
-              "Expected to migrate process instance with key '%d' to process definition with key '%d', but process definition not found",
-              value.getProcessInstanceKey(), targetProcessDefinitionKey));
+      final String reason =
```

Comment:
💭 I feel like the NOT_FOUND is reserved for the entity that is being migrated, so the process instance itself, rather than for an unknown process definition.

❓ @berkaycanbc What do you think this should be?
- NOT_FOUND because the process definition cannot be found
- INVALID_ARGUMENT because the user used an unknown process definition key
- FAILED_PRECONDITION because the process definition is unknown

Reply:
@korthout this should still be `NOT_FOUND` since there is a key of a resource given and we cannot find it in our system. But, if we need to have separate codes for them process instance and definition, `FAILED_PRECONDITION` can be a bit better than `INVALID_ARGUMENT` because the input is not malformed.

I would still go for `NOT_FOUND` to truly reflect the meaning. WDYT?